src/quantity.rs

```diff
@@ -210,6 +210,54 @@ macro_rules! quantity {
             impl super::Conversion<V> for super::$unit {})+
         }
 
+        #[cfg(feature = "std")]
+        mod parse {
+            use $crate::lib::str::FromStr;
+
+            /// Represents an error encountered while parsing a string into a `Quantity`.
+            #[derive(Clone, Copy, Debug)]
+            pub enum ParseError {
+                /// No separators (spaces) were encountered.
+                NoSeparator,
+                /// An error occured while parsing the value (first) portion of the string.
+                ///
+                /// Due to exhaustiveness and type system limitations, this variant does not encode
+                /// the underlying parse error.
+                ValueParseError,
+                /// The abbreviation used wasn't found for this quantity.
+                UnknownAbbreviation,
+            }
+
+            impl<U, V> FromStr for super::$quantity<U, V>
+            where
+                U: super::super::Units<V> + ?Sized,
+                V: $crate::num::Num + $crate::Conversion<V> + FromStr,
+                $(
+                    super::$unit: $crate::Conversion<V, T = V::T>,
```

Comment:
I see what you mean about this being ugly. What is the error if removed?

Reply:
One isolated error:
```
error[E0277]: the trait bound `si::volume_rate::register_ton_per_second: Conversion<V>` is not satisfied
   --> src/quantity.rs:307:49
    |
307 |                               $abbreviation => Ok(Self::new::<super::$unit>(value)),
    |                                                   ^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Conversion<V>` is not implemented for `si::volume_rate::register_ton_per_second`
    |
   ::: src/si/volume_rate.rs:3:1
    |
3   | / quantity! {
4   | |     /// Volume rate (base unit cubic meter per second, m<sup>3</sup> · s<sup>-1</sup>).
5   | |     quantity: VolumeRate; "volume rate";
6   | |     /// Volume dimension, m<sup>3</sup> · s<sup>-1</sup>.
...   |
108 | |     }
109 | | }
    | |_- in this macro invocation
    |
    = help: consider adding a `where si::volume_rate::register_ton_per_second: Conversion<V>` bound
note: required by `si::volume_rate::<impl si::Quantity<si::Dimension<L=typenum::PInt<typenum::UInt<typenum::UInt<typenum::UTerm, typenum::B1>, typenum::B1>>, M=typenum::Z0, T=typenum::NInt<typenum::UInt<typenum::UTerm, typenum::B1>>, Th=typenum::Z0, J=typenum::Z0, N=typenum::Z0, I=typenum::Z0> + 'static, U, V>>::new`
   --> src/quantity.rs:329:13
    |
329 | /             pub fn new<N>(v: V) -> Self
330 | |             where
331 | |                 N: Unit + $crate::Conversion<V, T = V::T>,
332 | |             {
...   |
337 | |                 }
338 | |             }
    | |_____________^
    |
   ::: src/si/volume_rate.rs:3:1
    |
3   | / quantity! {
4   | |     /// Volume rate (base unit cubic meter per second, m<sup>3</sup> · s<sup>-1</sup>).
5   | |     quantity: VolumeRate; "volume rate";
6   | |     /// Volume dimension, m<sup>3</sup> · s<sup>-1</sup>.
...   |
108 | |     }
109 | | }
    | |_- in this macro invocation
```

[Full error](https://gist.github.com/Aehmlo/e6ef84a319bc23cc27ae417e8f403928) (huge due to macro invocations).

As I mentioned, I'm sure there are better ways/workarounds to do this.